changelog/8-1-1-Release.md

```diff
@@ -0,0 +1,1048 @@
+    @Override
+    public String getDisplayName() {
+        return "Find exceeds resource ratio";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new YamlIsoVisitor<ExecutionContext>() {
+            @Override
+            public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, ExecutionContext ctx) {
+                Yaml.Mapping.Entry e = super.visitMappingEntry(entry, ctx);
+                return e.withMarkers(e.getMarkers().searchResult("foo"));
+            }
+        };
+    }
+}
+```
+{% endtab %}
+
+{% tab title="After" %}
+```java
+package org.openrewrite.kubernetes.resource;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.marker.SearchResult;
+import org.openrewrite.yaml.YamlIsoVisitor;
+import org.openrewrite.yaml.tree.Yaml;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class FindExceedsResourceRatio extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Find exceeds resource ratio";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new YamlIsoVisitor<ExecutionContext>() {
+            @Override
+            public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, ExecutionContext ctx) {
+                Yaml.Mapping.Entry e = super.visitMappingEntry(entry, ctx);
+                return SearchResult.found(e, "foo");
+            }
+        };
+    }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+### The doNext method has been removed
+
+The [org.openrewrite.Recipe doNext(..)](https://github.com/openrewrite/rewrite/blob/v7.40.8/rewrite-core/src/main/java/org/openrewrite/Recipe.java#L306-L313) method has been removed. If a recipe won't make any changes, this should be replaced with a [ScanningRecipe](#new-approach-to-recipe-visiting). If it does make changes, it should use [TreeVisitor#doAfterVisit()](https://github.com/openrewrite/rewrite/blob/v7.40.8/rewrite-core/src/main/java/org/openrewrite/TreeVisitor.java#L146).
+
+TODO: This migration recipe does not provide enough context. It links to a non-existant guide: https://github.com/openrewrite/rewrite/blob/main/rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MigrateRecipeToRewrite8Test.java#L485
+
+## Migrating your recipes
+
+We've created a [migration recipe](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/recipes/MigrateRecipeToRewrite8.java) that will assist you with migrating your recipes to the latest version. You should run this recipe against all of your existing recipes. If the migration recipe is not able to fully migrate your recipe, comments will be added to the code to request a human to review it.
+
+Please feel free to reach out in the [community Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA) if you have any questions.
+
+## New recipes
```

Comment:
You can ignore everything from here down -- this is auto-generated from the markdown generator.